pkg/log/errors.go

```diff
@@ -0,0 +1,44 @@
+package log
+
+// ErrorCategory defines the category of a pipeline error
+type ErrorCategory int
+
+// Error categories which allow categorizing failures
+const (
+	ErrorUndefined ErrorCategory = iota
+	ErrorBuild
+	ErrorCompliance
+	ErrorConfiguration
+	ErrorCustom
+	ErrorInfrastructure
+	ErrorService
+	ErrorTest
+)
+
+var errorCategory ErrorCategory = ErrorUndefined
```

Comment:
Just curious: Would `errorCategory := ErrorUndefined` work as well?

Reply:
I had issues with that on a global level some while back thus I have not tried ...